ext/standard/formatted_print.c

```diff
@@ -524,7 +524,7 @@ php_formatted_print(zval *z_format, zval *args, int argc)
 
 			if (argnum >= argc) {
 				efree(result);
-				php_error_docref(NULL, E_WARNING, "Too few arguments");
+				zend_value_error("Too few arguments");
```

Comment:
I apologise I forgot we had ``zend_argument_count_error()`` which probably makes more sense in this case.

Reply:
I started working on replacing the error message with something like:

```php
zend_argument_count_error("%d parameters are required, %d given", argnum + 2, argc + 1);
```

It makes a lot of sense most of the time, but it turns out I'm running in an issue with "vprintf".

vprintf takes an array of placeholders as the second argument:

```
vprintf ( string $format , array $args ) : int
```

The message generated would be something like:

`5 parameters are required, 3 given`

which is blatantly misleading.

I was thinking about replacing the message with something like:

`4 placeholders are required, 2 given`

My question: does it still qualifies as a `ArgumentCountError` ? Or should I stick to the `ValueError` ?

Reply:
I'm not familiar with this piece of code, but if the v*print* family of functions use the same handler it's probably best to keep a value error, (splitting the function in two seems a bit pointless here), because when using an array you are passing the correct number of arguments but the array has the wrong "shape", thus I'd use a ValueError in that case, if this means using it for all other cases, I would suppose this to be alright.

Reply:
I added a new arguments to `php_formatted_print` to specify the kind of error message to be thrown. That way, I can throw a `ValueError` for v*print* functions and a `ArgumentCountError` otherwise, with appropriate error messages.

I'm not completely sure if this make sense, and the error message (`"The arguments array must contain %d items, %d given"`) might need to be worked upon.

Let me know!

Reply:
I'm not sure either, tbh, @nikic any opinion? I don't feel like this complication is worth it, but the error messages being more descriptive is quite nice ...